test/compiler/immutablearray.jl

```diff
@@ -0,0 +1,459 @@
+using Test
```

Comment:
Okay, I updated this PR with the latest version of EA, and also merged the latest changes of `master`.
Now we should work on optimizations and try to pass these test cases (or mark some of them broken and leave them for later).